File: scripts/generate.py
#!/usr/bin/env python3
"""
OpenDoor Image - AI 图片生成与归档

支持两种协议：
- Gemini 格式（NanoBanana 系列）: POST /v1beta/models/{model}:generateContent
- OpenAI Images 格式（gpt-image-2）: POST /v1/images/generations

从环境变量读取配置:
- OPENDOOR_IMAGE_API_KEY: API 密钥 (必需)
- OPENDOOR_IMAGE_API_BASE: API 基础地址 (默认: https://api.code-opendoor.com)
- OPENDOOR_IMAGE_MODEL: 模型名称 (默认: gemini-3.1-flash-image)
- OPENDOOR_IMAGE_OUTPUT_DIR: 输出目录 (默认: ~/generated_images)
- OPENDOOR_IMAGE_SIZE: 图片尺寸，仅 gpt-image-2 有效 (默认: 1024x1024)
- OPENDOOR_IMAGE_QUALITY: 图片质量，仅 gpt-image-2 有效 (默认: low)

使用方法:
    python3 generate.py "一只可爱的猫"
    python3 generate.py "夕阳下的海边" --model gpt-image-2
    python3 generate.py --list
"""

import os
import sys
import json
import base64
import uuid
import re
import argparse
from datetime import datetime
from pathlib import Path
from typing import Optional

try:
    import requests
except ImportError:
    print("需要安装 requests: pip install requests")
    sys.exit(1)


def _find_env_file() -> Optional[Path]:
    """按优先级查找 .env 文件：skill 安装目录 > 项目目录"""
    candidates = [
        Path.home() / ".claude" / "skills" / "opendoor-image-skills" / ".env",
        Path(__file__).resolve().parent.parent / ".env",
    ]
    for p in candidates:
        if p.is_file():
            return p
    return None


def _parse_env_file(env_path: Path) -> dict[str, str]:
    """解析 .env 文件，忽略注释和空行"""
    result = {}
    with open(env_path, "r", encoding="utf-8") as f:
        for line in f:
            line = line.strip()
            if not line or line.startswith("#"):
                continue
            if "=" not in line:
                continue
            key, _, value = line.partition("=")
            key = key.strip()
            value = value.strip().strip('"').strip("'")
            if key and value:
                result[key] = value
    return result


def load_config() -> dict:
    env_file = _find_env_file()
    file_vars = _parse_env_file(env_file) if env_file else {}

    def _get(key: str, default: str = "") -> str:
        """环境变量优先，.env 文件次之，最后用默认值"""
        return os.getenv(key) or file_vars.get(key) or default

    config = {
        "api_key": _get("OPENDOOR_IMAGE_API_KEY"),
        "api_base": _get("OPENDOOR_IMAGE_API_BASE", "https://api.code-opendoor.com").rstrip("/"),
        "model": _get("OPENDOOR_IMAGE_MODEL", "gemini-3.1-flash-image"),
        "size": _get("OPENDOOR_IMAGE_SIZE", "1024x1024"),
        "quality": _get("OPENDOOR_IMAGE_QUALITY", "low"),
    }
    if not config["api_key"]:
        hint = "请在以下任一位置配置:\n"
        hint += f"  1. {Path.home() / '.claude' / 'skills' / 'opendoor-image-skills' / '.env'}\n"
        hint += f"  2. 环境变量 OPENDOOR_IMAGE_API_KEY\n"
        hint += "  获取密钥: https://api.code-opendoor.com"
        raise ValueError(f"未找到 OPENDOOR_IMAGE_API_KEY\n{hint}")
    return config


def is_openai_image_model(model: str) -> bool:
    return model.startswith("gpt-image") or model.startswith("dall-e")


def sanitize_filename(text: str, max_len: int = 20) -> str:
    text = re.sub(r'[^a-zA-Z0-9一-鿿㐀-䶿_]+', '_', text)
    result = text.strip('_')[:max_len].strip('_')
    return result or "untitled"


def _calc_next_sequence(index: dict, date_str: str) -> int:
    max_seq = 0
    for entry in index.values():
        fname = entry.get("filename", "")
        m = re.match(r'^\d{8}_(\d{3})_', fname)
        if m and fname.startswith(date_str):
            try:
                max_seq = max(max_seq, int(m.group(1)))
            except ValueError:
                pass
    return max_seq + 1


def load_index(output_dir: Path) -> dict:
    index_file = output_dir / ".index.json"
    if index_file.exists():
        try:
            with open(index_file, 'r', encoding='utf-8') as f:
                return json.load(f)
        except (json.JSONDecodeError, OSError):
            return {}
    return {}


def save_index(output_dir: Path, index: dict) -> None:
    index_file = output_dir / ".index.json"
    tmp_file = index_file.with_suffix('.tmp')
    try:
        with open(tmp_file, 'w', encoding='utf-8') as f:
            json.dump(index, f, ensure_ascii=False, indent=2)
        tmp_file.replace(index_file)
    except OSError as e:
        print(f"保存索引失败: {e}")
        tmp_file.unlink(missing_ok=True)


def _lock_index(output_dir: Path):
    """返回 .index.lock 文件对象（已加锁），调用方负责解锁关闭"""
    import platform
    lock_path = output_dir / ".index.lock"
    f = open(lock_path, 'w', encoding='utf-8')
    if platform.system() == "Windows":
        import msvcrt
        msvcrt.locking(f.fileno(), msvcrt.LK_LOCK, 1)
    else:
        import fcntl
        fcntl.flock(f, fcntl.LOCK_EX)
    return f


def _unlock_index(lock_file) -> None:
    import platform
    if platform.system() == "Windows":
        import msvcrt
        lock_file.seek(0)
        msvcrt.locking(lock_file.fileno(), msvcrt.LK_UNLCK, 1)
    else:
        import fcntl
        fcntl.flock(lock_file, fcntl.LOCK_UN)
    lock_file.close()


def _raise_for_status(resp) -> None:
    if not resp.ok:
        try:
            body = resp.json()
            msg = json.dumps(body, ensure_ascii=False)[:300]
        except Exception:
            msg = resp.text[:300]
        raise ValueError(f"API 返回 {resp.status_code}: {msg}")


def call_gemini(prompt: str, config: dict) -> tuple[bytes, str]:
    """调用 Gemini 格式接口（NanoBanana 系列）"""
    model = config["model"]
    url = f"{config['api_base']}/v1beta/models/{model}:generateContent"
    payload = {
        "contents": [{"parts": [{"text": prompt}]}],
        "generationConfig": {"responseModalities": ["TEXT", "IMAGE"]}
    }
    resp = requests.post(
        url,
        headers={
            "Authorization": f"Bearer {config['api_key']}",
            "Content-Type": "application/json"
        },
        json=payload,
        timeout=120
    )
    _raise_for_status(resp)
    data = resp.json()

    try:
        parts = data["candidates"][0]["content"]["parts"]
        for part in parts:
            if "inlineData" in part:
                return base64.b64decode(part["inlineData"]["data"]), part["inlineData"].get("mimeType", "image/png")
    except (KeyError, IndexError) as e:
        raise ValueError(f"解析 Gemini 响应失败: {e}\n响应: {json.dumps(data, ensure_ascii=False)[:500]}")

    raise ValueError(f"Gemini 响应中未找到图片数据\n响应: {json.dumps(data, ensure_ascii=False)[:500]}")


def call_openai_images(prompt: str, config: dict) -> tuple[bytes, str]:
    """调用 OpenAI Images 格式接口（gpt-image-2）"""
    url = f"{config['api_base']}/v1/images/generations"
    payload = {
        "model": config["model"],
        "prompt": prompt,
        "size": config["size"],
        "quality": config["quality"],
    }
    resp = requests.post(
        url,
        headers={
            "Authorization": f"Bearer {config['api_key']}",
            "Content-Type": "application/json"
        },
        json=payload,
        timeout=120
    )
    _raise_for_status(resp)
    data = resp.json()

    try:
        b64 = data["data"][0]["b64_json"]
        return base64.b64decode(b64), "image/png"
    except (KeyError, IndexError) as e:
        raise ValueError(f"解析 OpenAI Images 响应失败: {e}\n响应: {json.dumps(data, ensure_ascii=False)[:500]}")


def mime_to_ext(mime: str) -> str:
    known = {"image/png": "png", "image/jpeg": "jpg", "image/webp": "webp"}
    if mime not in known:
        print(f"警告: 未知 MIME 类型 {mime!r}，默认使用 png")
    return known.get(mime, "png")


def generate_image(
    prompt: str,
    output_dir: Optional[str] = None,
    model: Optional[str] = None,
    size: Optional[str] = None,
    quality: Optional[str] = None,
) -> dict:
    config = load_config()
    if model:
        config["model"] = model
    if size:
        config["size"] = size
    if quality:
        config["quality"] = quality

    if output_dir is None:
        out_path = Path(os.getenv("OPENDOOR_IMAGE_OUTPUT_DIR", str(Path.home() / "generated_images"))).expanduser()
    else:
        out_path = Path(output_dir).expanduser()
    out_path.mkdir(parents=True, exist_ok=True)

    print(f"正在生成: {prompt}")
    print(f"模型: {config['model']}")

    try:
        if is_openai_image_model(config["model"]):
            image_bytes, mime = call_openai_images(prompt, config)
        else:
            image_bytes, mime = call_gemini(prompt, config)
    except (requests.exceptions.RequestException, ValueError) as e:
        raise RuntimeError(f"生成失败: {e}") from e

    ext = mime_to_ext(mime)
    now = datetime.now()
    date_str = now.strftime("%Y%m%d")
    content_hash = uuid.uuid4().hex[:12]

    lock_file = _lock_index(out_path)
    try:
        index = load_index(out_path)
        seq = _calc_next_sequence(index, date_str)
        desc = sanitize_filename(prompt)
        filename = f"{date_str}_{seq:03d}_{desc}_{content_hash[:6]}.{ext}"
        target_path = out_path / filename

        with open(target_path, "wb") as f:
            f.write(image_bytes)

        created_at = now.isoformat()
        index[content_hash] = {
            "filename": filename,
            "prompt": prompt,
            "model": config["model"],
            "created_at": created_at,
            "path": str(target_path)
        }
        save_index(out_path, index)
    finally:
        _unlock_index(lock_file)

    size_kb = len(image_bytes) / 1024
    print(f"图片已保存: {target_path}")
    print(f"大小: {size_kb:.1f} KB")

    return {
        "path": str(target_path),
        "filename": filename,
        "prompt": prompt,
        "created_at": created_at,
        "size_kb": size_kb
    }


def list_images(output_dir: Optional[str] = None, limit: int = 20) -> None:
    if output_dir is None:
        out_path = Path(os.getenv("OPENDOOR_IMAGE_OUTPUT_DIR", str(Path.home() / "generated_images"))).expanduser()
    else:
        out_path = Path(output_dir).expanduser()

    if not out_path.exists():
        print(f"目录不存在: {out_path}")
        return

    limit = min(limit, 200)

    # 优先从 .index.json 读取（含 prompt/model 元数据）
    index = load_index(out_path)
    if index:
        entries = sorted(index.values(), key=lambda x: x.get("created_at", ""), reverse=True)
        print(f"\n生成的图片 (共 {len(entries)} 张):\n")
        for entry in entries[:limit]:
            path = Path(entry.get("path", ""))
            size_kb = path.stat().st_size / 1024 if path.exists() else 0
            created = entry.get("created_at", "")[:16].replace("T", " ")
            prompt_short = entry.get("prompt", "")[:30]
            model = entry.get("model", "")
            print(f"  {entry.get('filename', path.name)}")
            print(f"    提示词: {prompt_short}  模型: {model}  大小: {size_kb:.1f} KB  时间: {created}")
        if len(entries) > limit:
            print(f"\n... 还有 {len(entries) - limit} 张")
        return

    # fallback：直接遍历文件系统
    entries_fs = []
    for p in out_path.iterdir():
        if p.suffix in ('.jpg', '.png', '.webp'):
            try:
                entries_fs.append((p, p.stat()))
            except OSError:
                pass
    entries_fs.sort(key=lambda x: x[0].name, reverse=True)

    print(f"\n生成的图片 (共 {len(entries_fs)} 张):\n")
    for img, stat in entries_fs[:limit]:
        size_kb = stat.st_size / 1024
        created = datetime.fromtimestamp(stat.st_mtime).strftime("%Y-%m-%d %H:%M")
        print(f"  {img.name}  ({size_kb:.1f} KB | {created})")

    if len(entries_fs) > limit:
        print(f"\n... 还有 {len(entries_fs) - limit} 张")


def main() -> None:
    parser = argparse.ArgumentParser(description="OpenDoor Image - AI 图片生成与归档")
    parser.add_argument("prompt", nargs="?", help="图片生成提示词")
    parser.add_argument("--output", "-o", dest="output_dir", help="输出目录")
    parser.add_argument("--list", "-l", action="store_true", help="列出已生成的图片")
    parser.add_argument("--model", "-m", help="模型名称")
    parser.add_argument("--size", "-s", help="图片尺寸，仅 gpt-image-2 有效（如 1024x1024）")
    parser.add_argument("--quality", "-q", help="图片质量，仅 gpt-image-2 有效（low/medium/high）")
    args = parser.parse_args()

    try:
        if args.list:
            list_images(args.output_dir)
        elif args.prompt:
            generate_image(args.prompt, args.output_dir, args.model, args.size, args.quality)
        else:
            parser.print_help()
            sys.exit(1)
    except (ValueError, RuntimeError) as e:
        print(f"错误: {e}")
        sys.exit(1)


if __name__ == "__main__":
    main()

File: scripts/generate.js
#!/usr/bin/env node
/**
 * OpenDoor Image - AI 图片生成与归档 (Node.js)
 *
 * 支持两种协议：
 * - Gemini 格式（NanoBanana 系列）
 * - OpenAI Images 格式（gpt-image-2）
 *
 * 配置优先级：环境变量 > .env 文件 > 默认值
 */

import fs from "fs";
import path from "path";
import { randomUUID } from "crypto";
import { parseArgs } from "util";

// Node.js 18+ 原生 fetch，无需额外依赖

// ─── 配置加载 ────────────────────────────────────────────────

function findEnvFile() {
  const candidates = [
    path.join(process.env.HOME || process.env.USERPROFILE, ".claude", "skills", "opendoor-image-skills", ".env"),
    path.join(path.dirname(new URL(import.meta.url).pathname), "..", ".env"),
  ];
  return candidates.find((p) => fs.existsSync(p)) ?? null;
}

function parseEnvFile(envPath) {
  const vars = {};
  const lines = fs.readFileSync(envPath, "utf-8").split("\n");
  for (const raw of lines) {
    const line = raw.trim();
    if (!line || line.startsWith("#") || !line.includes("=")) continue;
    const idx = line.indexOf("=");
    const key = line.slice(0, idx).trim();
    const value = line.slice(idx + 1).trim().replace(/^["']|["']$/g, "");
    if (key && value) vars[key] = value;
  }
  return vars;
}

function loadConfig(overrides = {}) {
  const envPath = findEnvFile();
  const fileVars = envPath ? parseEnvFile(envPath) : {};

  const get = (key, def = "") =>
    overrides[key] || process.env[key] || fileVars[key] || def;

  const apiKey = get("OPENDOOR_IMAGE_API_KEY");
  if (!apiKey) {
    const hint = [
      "请在以下任一位置配置:",
      `  1. ${path.join(process.env.HOME || process.env.USERPROFILE, ".claude", "skills", "opendoor-image-skills", ".env")}`,
      "  2. 环境变量 OPENDOOR_IMAGE_API_KEY",
      "  获取密钥: https://api.code-opendoor.com",
    ].join("\n");
    throw new Error(`未找到 OPENDOOR_IMAGE_API_KEY\n${hint}`);
  }

  return {
    apiKey,
    apiBase: get("OPENDOOR_IMAGE_API_BASE", "https://api.code-opendoor.com").replace(/\/$/, ""),
    model: get("OPENDOOR_IMAGE_MODEL", "gemini-3.1-flash-image"),
    size: get("OPENDOOR_IMAGE_SIZE", "1024x1024"),
    quality: get("OPENDOOR_IMAGE_QUALITY", "low"),
  };
}

// ─── 工具函数 ────────────────────────────────────────────────

function isOpenAIModel(model) {
  return model.startsWith("gpt-image") || model.startsWith("dall-e");
}

function sanitizeFilename(text, maxLen = 20) {
  const result = text.replace(/[^a-zA-Z0-9\u4e00-\u9fff\u3400-\u4dbf_]+/g, "_")
    .replace(/^_+|_+$/g, "")
    .slice(0, maxLen)
    .replace(/^_+|_+$/g, "");
  return result || "untitled";
}

function mimeToExt(mime) {
  const known = { "image/png": "png", "image/jpeg": "jpg", "image/webp": "webp" };
  if (!known[mime]) console.warn(`警告: 未知 MIME 类型 ${mime}，默认使用 png`);
  return known[mime] ?? "png";
}

async function raiseForStatus(resp) {
  if (!resp.ok) {
    let msg;
    try {
      const body = await resp.json();
      msg = JSON.stringify(body).slice(0, 300);
    } catch {
      msg = (await resp.text()).slice(0, 300);
    }
    throw new Error(`API 返回 ${resp.status}: ${msg}`);
  }
}

// ─── 文件锁（跨平台，基于 .lock 文件 + 重试） ───────────────

function acquireLock(lockPath, timeout = 10000) {
  const start = Date.now();
  while (true) {
    try {
      fs.writeFileSync(lockPath, String(process.pid), { flag: "wx" });
      return;
    } catch (e) {
      if (e.code !== "EEXIST") throw e;
      if (Date.now() - start > timeout) throw new Error("获取文件锁超时");
      // 简单自旋等待
      const end = Date.now() + 50;
      while (Date.now() < end) {}
    }
  }
}

function releaseLock(lockPath) {
  try { fs.unlinkSync(lockPath); } catch {}
}

// ─── 索引管理 ────────────────────────────────────────────────

function loadIndex(outDir) {
  const indexPath = path.join(outDir, ".index.json");
  if (!fs.existsSync(indexPath)) return {};
  try {
    return JSON.parse(fs.readFileSync(indexPath, "utf-8"));
  } catch {
    return {};
  }
}

function saveIndex(outDir, index) {
  const indexPath = path.join(outDir, ".index.json");
  const tmpPath = indexPath + ".tmp";
  fs.writeFileSync(tmpPath, JSON.stringify(index, null, 2), "utf-8");
  fs.renameSync(tmpPath, indexPath);
}

function calcNextSequence(index, dateStr) {
  let max = 0;
  for (const entry of Object.values(index)) {
    const m = entry.filename?.match(/^\d{8}_(\d{3})_/);
    if (m && entry.filename.startsWith(dateStr)) {
      max = Math.max(max, parseInt(m[1], 10));
    }
  }
  return max + 1;
}

// ─── API 调用 ────────────────────────────────────────────────

async function callGemini(prompt, config) {
  const url = `${config.apiBase}/v1beta/models/${config.model}:generateContent`;
  const resp = await fetch(url, {
    method: "POST",
    headers: {
      Authorization: `Bearer ${config.apiKey}`,
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      contents: [{ parts: [{ text: prompt }] }],
      generationConfig: { responseModalities: ["TEXT", "IMAGE"] },
    }),
    signal: AbortSignal.timeout(120_000),
  });
  await raiseForStatus(resp);
  const data = await resp.json();

  const parts = data?.candidates?.[0]?.content?.parts ?? [];
  for (const part of parts) {
    if (part.inlineData) {
      return {
        bytes: Buffer.from(part.inlineData.data, "base64"),
        mime: part.inlineData.mimeType ?? "image/png",
      };
    }
  }
  throw new Error(`Gemini 响应中未找到图片数据\n${JSON.stringify(data).slice(0, 500)}`);
}

async function callOpenAIImages(prompt, config) {
  const url = `${config.apiBase}/v1/images/generations`;
  const resp = await fetch(url, {
    method: "POST",
    headers: {
      Authorization: `Bearer ${config.apiKey}`,
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      model: config.model,
      prompt,
      size: config.size,
      quality: config.quality,
    }),
    signal: AbortSignal.timeout(120_000),
  });
  await raiseForStatus(resp);
  const data = await resp.json();

  const b64 = data?.data?.[0]?.b64_json;
  if (!b64) throw new Error(`解析 OpenAI Images 响应失败\n${JSON.stringify(data).slice(0, 500)}`);
  return { bytes: Buffer.from(b64, "base64"), mime: "image/png" };
}

// ─── 主功能 ──────────────────────────────────────────────────

async function generateImage(prompt, opts = {}) {
  const config = loadConfig({
    OPENDOOR_IMAGE_MODEL: opts.model,
    OPENDOOR_IMAGE_SIZE: opts.size,
    OPENDOOR_IMAGE_QUALITY: opts.quality,
  });

  const outDir = path.resolve(
    (opts.outputDir || process.env.OPENDOOR_IMAGE_OUTPUT_DIR || path.join(process.env.HOME || process.env.USERPROFILE, "generated_images"))
      .replace(/^~/, process.env.HOME || process.env.USERPROFILE)
  );
  fs.mkdirSync(outDir, { recursive: true });

  console.log(`正在生成: ${prompt}`);
  console.log(`模型: ${config.model}`);

  let result;
  try {
    result = isOpenAIModel(config.model)
      ? await callOpenAIImages(prompt, config)
      : await callGemini(prompt, config);
  } catch (e) {
    throw new Error(`生成失败: ${e.message}`);
  }

  const { bytes, mime } = result;
  const ext = mimeToExt(mime);
  const now = new Date();
  const dateStr = now.toISOString().slice(0, 10).replace(/-/g, "");
  const contentHash = randomUUID().replace(/-/g, "");
  const lockPath = path.join(outDir, ".index.lock");

  acquireLock(lockPath);
  let filename, targetPath, createdAt;
  try {
    const index = loadIndex(outDir);
    const seq = calcNextSequence(index, dateStr);
    const desc = sanitizeFilename(prompt);
    filename = `${dateStr}_${String(seq).padStart(3, "0")}_${desc}_${contentHash.slice(0, 6)}.${ext}`;
    targetPath = path.join(outDir, filename);

    fs.writeFileSync(targetPath, bytes);

    createdAt = now.toISOString();
    index[contentHash] = { filename, prompt, model: config.model, created_at: createdAt, path: targetPath };
    saveIndex(outDir, index);
  } finally {
    releaseLock(lockPath);
  }

  const sizeKb = (bytes.length / 1024).toFixed(1);
  console.log(`图片已保存: ${targetPath}`);
  console.log(`大小: ${sizeKb} KB`);

  return { path: targetPath, filename, prompt, created_at: createdAt, size_kb: parseFloat(sizeKb) };
}

function listImages(opts = {}) {
  const outDir = path.resolve(
    (opts.outputDir || process.env.OPENDOOR_IMAGE_OUTPUT_DIR || path.join(process.env.HOME || process.env.USERPROFILE, "generated_images"))
      .replace(/^~/, process.env.HOME || process.env.USERPROFILE)
  );

  if (!fs.existsSync(outDir)) {
    console.log(`目录不存在: ${outDir}`);
    return;
  }

  const limit = Math.min(opts.limit ?? 20, 200);
  const index = loadIndex(outDir);

  if (Object.keys(index).length > 0) {
    const entries = Object.values(index).sort((a, b) =>
      (b.created_at ?? "").localeCompare(a.created_at ?? "")
    );
    console.log(`\n生成的图片 (共 ${entries.length} 张):\n`);
    for (const entry of entries.slice(0, limit)) {
      let sizeKb = 0;
      try { sizeKb = (fs.statSync(entry.path).size / 1024).toFixed(1); } catch {}
      const created = (entry.created_at ?? "").slice(0, 16).replace("T", " ");
      console.log(`  ${entry.filename}`);
      console.log(`    提示词: ${(entry.prompt ?? "").slice(0, 30)}  模型: ${entry.model}  大小: ${sizeKb} KB  时间: ${created}`);
    }
    if (entries.length > limit) console.log(`\n... 还有 ${entries.length - limit} 张`);
    return;
  }

  // fallback：遍历文件系统
  const files = fs.readdirSync(outDir)
    .filter((f) => /\.(png|jpg|webp)$/.test(f))
    .map((f) => ({ name: f, stat: fs.statSync(path.join(outDir, f)) }))
    .sort((a, b) => b.name.localeCompare(a.name));

  console.log(`\n生成的图片 (共 ${files.length} 张):\n`);
  for (const { name, stat } of files.slice(0, limit)) {
    const sizeKb = (stat.size / 1024).toFixed(1);
    const created = new Date(stat.mtimeMs).toISOString().slice(0, 16).replace("T", " ");
    console.log(`  ${name}  (${sizeKb} KB | ${created})`);
  }
  if (files.length > limit) console.log(`\n... 还有 ${files.length - limit} 张`);
}

// ─── CLI 入口 ────────────────────────────────────────────────

const { values, positionals } = parseArgs({
  args: process.argv.slice(2),
  options: {
    model: { type: "string", short: "m" },
    output: { type: "string", short: "o" },
    list: { type: "boolean", short: "l", default: false },
    size: { type: "string", short: "s" },
    quality: { type: "string", short: "q" },
  },
  allowPositionals: true,
});

try {
  if (values.list) {
    listImages({ outputDir: values.output });
  } else if (positionals.length > 0) {
    await generateImage(positionals[0], {
      model: values.model,
      outputDir: values.output,
      size: values.size,
      quality: values.quality,
    });
  } else {
    console.log("用法: node generate.js <提示词> [--model <模型>] [--output <目录>]");
    console.log("      node generate.js --list");
    process.exit(1);
  }
} catch (e) {
  console.error(`错误: ${e.message}`);
  process.exit(1);
}
